services/SeomaticService.php

```diff
@@ -2568,12 +2566,11 @@ protected function getOffersArrayForJsonLD(BaseElementModel $element, array $ide
     {
         $offers = [];
 
-        if (isset($element->productOffers) && isset($element->productCurrency))
-        {
-            foreach($element->productOffers as $offer)
-            {
-                if (isset($offer->offerName) && isset($offer->offerPrice))
-                {
+        if (isset($element->productOffers) && isset($element->productCurrency)) {
+
+            foreach($element->productOffers as $offer) {
+
+                if (isset($offer->offerName) && isset($offer->offerPrice)) {
```

Comment:
more scope for early returns here...
```php
if (!isset($element->productOffers) || !isset($element->productCurrency)) {
    return $offers;
}

foreach($element->productOffers as $offer) {
        if (!isset($offer->offerName) || !isset($offer->offerPrice)) {
            continue;
        }

        ...
```

Reply:
I was told that it's better to have one return in a function during my studies, that's why I've got some trouble to get it done as you told me ;-)